mappings/loc.py

```diff
@@ -8,32 +9,98 @@ def __init__(self, source):
     def createMapping(self):
         return {
             'title': ('title', '{0}'),
-            'alternate': [('other_title', '{0}')], #One other_title in items block and one outside of it 
-            'medium': [('medium', '{0}')],
-            'authors': [('contributor', '{0}|||true')],
-            'languages': [('language', '{0}||')],
-            'dates': [('dates', '{0}|publication_date')], #Apply formatting because dates in a list
-            'publisher': [('created_published', '{0}')],
+            'alternative': [('other_title', '{0}')], #One other_title in items block and one outside of it 
+            'medium': [('original_format', '{0}')],
+            'authors': ('contributor', '{0}|||true'),
+            'languages': ('item', '||{0}'),
+            'dates': ('dates', '{0}|publication_date'),
+            'publisher': ('item', '{0}'),
             'identifiers': [
                 ('number_lccn', '{0}|loc'),
-                ('call_number', '{0}|callNumber'),
+                ('item', '{0}'),
             ],
-            'contributors': [
-                ('contributors', '{0}|||contributor'),
-            ],
-            'extent': [('medium', '{0}')],
-            'is_part_of': [('partof', '{0}|collection')],
-            'spatial': ('created_published', '{0}'),#publisherLocation to be in apply formatting
-            'abstract': [
+            'contributors': 
+                ('contributor', '{0}|||contributor'),
+            'extent': [('item', '{0}')],
+            'is_part_of': ('partof', '{0}|collection'),
+            'abstract': 
                 ('description', '{0}')
-            ],
-            'subjects': [('subjects', '{0}||')],
-            'rights': (['rights_advisory', 'copyra', 'copyri'], 'met|{0}|{1}|{2}|'),
-            'has_part': [('pdf', '1|{0}|loc|application/pdf|{{"catalog": false, "download": true, "reader": false, "embed": false}}')]
+            ,
+            'subjects': [('item', '{0}')],
         }
 
     def applyFormatting(self):
+        self.record.has_part = []
         self.record.source = 'loc'
-        self.record.source_id = self.record.identifiers[0]
         self.record.medium = self.record.medium[0]
-        self.record.spatial = self.record.spatial.split(':')[0].strip(' ')
+
+        #Convert string repr of list to actual list
+        itemList = ast.literal_eval(self.record.identifiers[1])
```

Comment:
hmmm, what does `self.record.identifiers` look like here? If it's just a json string, I think it would be better to do a `json.loads`, would be a bit clearer / more standard. Unless the format of the string is not valid json and is something else?

Reply:
I mapped the `items` block to `self.record.identifiers[1]` and it comes in as a string representation of a dictionary with single quotes while a valid JSON strings needs to be in double quotes. Thus, I get the `Expecting property name enclosed in double quotes` error when using `json.loads`. I could use` json.dumps` then use `json.loads` but that's extra lines of code compared to the one with the `ast.literal_eval` method I'm currently using.

Reply:
hmm, I see. One follow up, when I throw that string into the `literal_eval` method, it seems to not like it:
```
>>> ast.literal_eval("['identifier': 'test']")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ast.py", line 62, in literal_eval
    node_or_string = parse(node_or_string.lstrip(" \t"), mode='eval')
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    ['identifier': 'test']
                 ^
SyntaxError: invalid syntax
```

is that the right string? And could you maybe add a comment describing the string here so it's more clear why we're using `literal_eval`?

Reply:
It's supposed to be a dictionary like "{'identifier': "test'}". If you put curly braces, then it should work for you.